# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerDebuggerManagerChild.cpp
# Commit: e80b18ca9165
# Full Hash: e80b18ca916518c73d5435fe8f842d9f0e1b7585
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-25 21:02:33
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r cb743703a9ee -r e80b18ca9165 dom/workers/remoteworkers/RemoteWorkerDebuggerManagerChild.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerDebuggerManagerChild.cpp	Tue Mar 25 09:13:20 2025 +0000
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "RemoteWorkerDebuggerManagerChild.h"
+#include "RemoteWorkerService.h"
+
+namespace mozilla::dom {
+
+RemoteWorkerDebuggerManagerChild::RemoteWorkerDebuggerManagerChild() {
+  MOZ_ASSERT_DEBUG_OR_FUZZING(
+      RemoteWorkerService::Thread() &&
+      RemoteWorkerService::Thread()->IsOnCurrentThread());
+}
+
+RemoteWorkerDebuggerManagerChild::~RemoteWorkerDebuggerManagerChild() {
+  MOZ_ASSERT_DEBUG_OR_FUZZING(
+      RemoteWorkerService::Thread() &&
+      RemoteWorkerService::Thread()->IsOnCurrentThread());
+}
+
+}  // end of namespace mozilla::dom